src/parser.js

```diff
@@ -246,16 +247,29 @@ function createCreateStatementParser (isStrict) {
         acceptTokens: [
           { type: 'keyword', value: 'TABLE' },
           { type: 'keyword', value: 'DATABASE' },
+          { type: 'keyword', value: 'TRIGGER' },
         ],
       },
       add: (token) => {
         statement.type = `CREATE_${token.value.toUpperCase()}`;
       },
       postCanGoToNext: () => true,
     },
+    // {
+    //   preCanGoToNext: () => false,
+    //   validation: {
+    //     acceptTokens: [
+    //       { type: 'keyword', value: 'END' },
+    //     ],
+    //     add: (token) => {
+    //       statement.endStatement = 'END'
+    //     }
+    //   },
+    //   postCanGoToNext: () => false,
+    // },
   ];
 
-  return stateMachineStatementParser(isStrict, statement, steps);
+  return stateMachineStatementParser(isStrict, statement, steps, dialect);
```

Comment:
Looking at this PR and the functions in this file, I think we should unify on a consistent ordering of things as we have three different orderings of where `isStrict` and `dialect` go in the function list:

* `parse(query, isStrict, dialect)` (both last, dialect immediately follows isStrict)
* `createStatementParserByToken (isStrict, dialect, token)` (both first, dialect immediately follows isStrict)
* `stateMachineStatementParser(isStrict, statement, steps, dialect)` (isStrict is first, dialect last)

I would advocate for `(isStrict, dialect, ...args)` as this would be the easiest to follow consistently I think with the same two first args in all cases. I do not think we necessarily need to do this refactoring here beyond maybe unifying the arg setup of `createStatementParserByToken` and `stateMachineStatementParser`.

Reply:
I actually don't like strict and dialect at the beginning because it makes it hard to extend. What if we added `version` or some other thing in future?

I'd actually prefer `(args, options)`, where options currently includes isStrict and dialect.

I'll do that tidy-up now.